cmd/autoheal/healer.go

```diff
@@ -147,8 +147,17 @@ func (b *HealerBuilder) Build() (h *Healer, err error) {
 	h.rulesQueue = workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "rules")
 	h.alertsQueue = workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "alerts")
 
-	// initialize active jobs map:
-	h.activeJobs = new(syncmap.Map)
+	h.jobRunners = map[JobRunnerType]JobRunner{
```

Comment:
If we are using an interface to hide the implementation details of each job runner, then we should also hide the implementation details of the creation of the job runner, we should not access directly internal fields like `config` or `activeJobs`. To enforce this I'd suggest that you move the implementations of the job runners to different packages, and make everything private, unless really needed. That will force you to create a clean interface between the healer and the job runner.

Reply:
Working on it :+1: